zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/distribution/CommandDistributionBehavior.java

```diff
@@ -149,7 +160,39 @@ public <T extends UnifiedRecordValue> void distributeCommand(
         (partition) -> distributeToPartition(partition, distributionRecord, distributionKey));
   }
 
-  private <T extends UnifiedRecordValue> void distributeToPartition(
+  private void distributeToPartition(
+      final int partition,
+      final CommandDistributionRecord distributionRecord,
+      final long distributionKey) {
+    final var distributionQueue = Optional.ofNullable(distributionRecord.getQueueId());
+    distributionQueue.ifPresent(queue -> enqueueDistribution(queue, partition, distributionKey));
+
+    final var otherQueuedDistributions =
```

Comment:
🔧 I find this name a bit miselading as it seems like it's getting a collection of other distributions. Reality is that it will return a an Optional containing the next distributionKey. How about renaming it to something like `nextDistributionKeyOptional` or something of the sorts?

Reply:
I'm doing the `isEmpty` check immediately now so that the boolean variable can be named `canDistributeImmediately`.